library/hls/src/main/java/com/google/android/exoplayer2/source/hls/HlsSampleStreamWrapper.java

```diff
@@ -652,7 +652,15 @@ public boolean continueLoading(long positionUs) {
 
   @Override
   public void reevaluateBuffer(long positionUs) {
-    // Do nothing.
+    if (loader.isLoading() || isPendingReset()) {
+      return;
+    }
+
+    int currentQueueSize = mediaChunks.size();
+    int preferredQueueSize = chunkSource.getPreferredQueueSize(positionUs, readOnlyMediaChunks);
+    if (currentQueueSize <= preferredQueueSize) {
+      return;
+    }
```

Comment:
Just the first checkin, see next changes to the class for the balance of the code.  It seems extending `ChunkSampleStream` is a better path for `HlsSampleStreamWrapper`.  Again, if that is the desired architecture I'm fine with taking that path.